polaris-core/src/main/java/io/polaris/core/persistence/ResolvedPolarisEntity.java

```diff
@@ -45,6 +46,7 @@ public ResolvedPolarisEntity(
   }
 
   public ResolvedPolarisEntity(EntityCacheEntry cacheEntry) {
+    Preconditions.checkNotNull(cacheEntry, "An entity cache can't be null.");
```

Comment:
The check is redundant, because recent JVMs throw meaningful NPEs telling that `cacheEntry` is `null` - the proposed message doesn't give any advice to users.